connectors/kafka/src/test/java/forklift/integration/ConsumerActionTests.java

```diff
@@ -0,0 +1,40 @@
+package forklift.integration;
+
+import forklift.Forklift;
+import forklift.connectors.ConnectorException;
+import forklift.consumer.Consumer;
+import forklift.exception.StartupException;
+import forklift.producers.ForkliftProducerI;
+import forklift.producers.ProducerException;
+import org.junit.Test;
+
+public class ConsumerActionTests extends BaseIntegrationTest {
+
+    /**
+     * Test subscribing to a topic, unsubscribing, then resubscribing to the topic.
+     */
+    @Test
+    public void topicTurnoverTest() throws StartupException, ConnectorException, InterruptedException, ProducerException {
+        Forklift forklift = serviceManager.newManagedForkliftInstance();
+        ForkliftProducerI producer = forklift.getConnector().getQueueProducer("forklift-string-topic");
+        sentMessageIds.add(producer.send("message1"));
+        final Consumer c1 = new Consumer(StringConsumer.class, forklift);
+        // Shutdown the consumer after all the messages have been processed.
+        c1.setOutOfMessages((listener) -> {
+            listener.shutdown();
+        });
+        // Start the consumer.
+        c1.listen();
+        Thread.sleep(3000);
```

Comment:
Thinking some more, I guess I'd blame this being weird on `Consumer.shutdown` being a signal rather than an synchronous action that completes when the consumer is truly "shutdown" or a asynchronous task with a `CompletableFuture`. A better name for what it's doing is `signalShutdown`. :man_shrugging: